### PR TITLE
Coco/dropdownon blur

### PR DESCRIPTION
**Pull request by ycoco**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ npm run change`

#### Description of changes
1. add onBlur event for dropDown 
2. rename autoComplete -> autoCompleteInput in ComboBox

(give an overview)

#### Focus areas to test

(optional)

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.Props.ts [hunk 1]
```diff
@@ -45,7 +45,7 @@ export interface IComboBoxProps extends ISelectableDroppableTextProps<IComboBox>
    * Whether the ComboBox auto completes. As the user is inputing text, it will be suggested potential matches from the list of items. If
    * the combo box is expanded, this will also scroll to the suggested item, and give it a selected style. Defaults to false.
    */
-  autoComplete?: boolean;
+  autoCompleteInput?: boolean;
```
Comment (micahgodbolt) [inline packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.Props.ts]: We need to deprecate the old value and give developers time to convert their code over to the new prop. Give me a shout if you need a hand with that.

Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.Props.ts]: I don't think we can in this case. The term `autoComplete` IS actually defined on the HTMLAttribute interface, so technically it is still there, but its type has changed. `autoComplete` is normally a string `off` or `on` I think. We unfortunately hit this break in adding the `HTMLAttribute` inheritance, but we need that to inject html attributes into the element. The code was just added last week, I think we have to make this an exception.

Comment (joschect) [inline packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.Props.ts]: You could also change the places where it could be used to check for `autoComplete && autoComplete ==="on"`

Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.Props.ts]: Actually that might be a better change. @ycoco would you mind trying that?

Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.Props.ts]: It might be worthwhile to leave the autoComplete setting defined in the interface so that it gets documented. It's an inconsistent attribute since it's not just a boolean.

File: packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.Props.ts [hunk 2]
```diff
@@ -45,7 +45,7 @@ export interface IComboBoxProps extends ISelectableDroppableTextProps<IComboBox>
    * Whether the ComboBox auto completes. As the user is inputing text, it will be suggested potential matches from the list of items. If
    * the combo box is expanded, this will also scroll to the suggested item, and give it a selected style. Defaults to false.
    */
-  autoComplete?: boolean;
+  autoComplete?: string;
```
Comment (cschleiden) [inline packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.Props.ts]: Why do we change to string? This is breaking change, can't we keep `boolean` and only map to `on` once we actually render?

File: packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx [hunk 3]
```diff
@@ -420,7 +425,27 @@ export class Dropdown extends BaseComponent<IDropdownInternalProps, IDropdownSta
...
+  private _onDropdownBlur(ev: React.FocusEvent<HTMLDivElement>) {
+    if (this.state.isOpen) {
+      // Do not onBlur when the callout is opened
+      return;
+    }
+    if (this.props.onBlur) {
+      this.props.onBlur(ev);
+      if (ev.preventDefault) {
+        return;
+      }
```
Comment (cschleiden) [inline packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx]: why is that required?

Comment (ycoco) [inline packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx]: Because we can't do onBlur on dropdown right now.  In client form, we have display and edit modes.  If you click the field it will change to edit mode, and then click another place that is not in drop down (even if no selection has been made), it should go back to display mode.  However, there is no way to go back to display mode without making a selection right now.

Comment (cschleiden) [inline packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx]: No, I mean the `return` if `preventDefault` is true. It will return in any case?

Comment (ycoco) [inline packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx]: oh, sorry.  If users want to overwrite them, they can say event.prevenDefault() and it will only run their code and won't run ours.

Comment (cschleiden) [inline packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx]: Yeah, but there is no "ours" here?

File: packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.Props.ts (hunk 2 above)
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.Props.ts]: @cschleiden I'm unsure of how to not break. Because, `autoComplete` is on HTMLAttributes as a strong, so either we don't use HTMLAttributes or we use string to align.

Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.Props.ts]: Given that the component was just introduced, it is probably safe to make this change. Other scenarios are broken because the props don't subclass from HTMLAttributes.
